### PR TITLE
[Cloud Run] Identity platform sample

### PR DESCRIPTION
**Pull request by averikitsch**
For tutorial: https://cloud.google.com/run/docs/tutorials/identity-platform
Waiting for PR #4495 for Kokoro variables

> It's a good idea to open an issue first for discussion.

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by eaball35** [comment]
Screen Shot 2021-01-05 at 9 08 09 AM
Screen Shot 2021-01-05 at 9 08 42 AM

**Comment by averikitsch** [comment]
Ignoring the inclusive language due to the branch `master` being in the link

File: run/idp-sql/app.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (kurtisvg) [inline run/idp-sql/app.json:1]: What's this file used for? 

Should we include a header in it?

Comment (averikitsch) [reply run/idp-sql/app.json:1]: This is for the Cloud Run button to build the sample correctly. Unfortunately I can only specify in JSON.

File: run/idp-sql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/idp-sql/pom.xml:1]: done

File: run/idp-sql/pom.xml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-dependencies</artifactId>
+				<version>2.4.1</version>
+				<type>pom</type>
+				<scope>import</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-dependencies</artifactId>
+				<version>${spring-cloud.version}</version>
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/idp-sql/pom.xml:49]: done

File: run/idp-sql/src/main/java/com/example/cloudrun/IdpSqlApplication.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    // Flush async logs if needed
+    // Current Logback config defaults to immediate flushing of all logs
+  }
+
+  /** Retrieve project Id from metadata server Set $GOOGLE_CLOUD_PROJECT env var to run locally */
+  public static String getProjectId() {
+    OkHttpClient ok =
+        new OkHttpClient.Builder()
+            .readTimeout(500, TimeUnit.MILLISECONDS)
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/idp-sql/src/main/java/com/example/cloudrun/IdpSqlApplication.java:80]: I've used this timeout for a few samples and never had an issue. Is that enough evidence?

File: run/idp-sql/src/main/resources/logback-spring.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2019 Google LLC
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply run/idp-sql/src/main/resources/logback-spring.xml]: done

File: run/idp-sql/src/main/java/com/example/cloudrun/IdpSqlApplication.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (kurtisvg) [reply run/idp-sql/src/main/java/com/example/cloudrun/IdpSqlApplication.java:80]: I would recommend more like 1-2 seconds, as some metadata servers don't always respond when first starting up (I know I occasionally this this problem on GKE, but maybe isn't an issue for RUN). 

I'll leave it up to you.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**